filters/json.js

```diff
@@ -0,0 +1,62 @@
+// i18next j4 json format defined at https://www.i18next.com/misc/json-format
+import flatten from "flat";
+
+export class JsonFilter {
+    constructor(params) {
+        this.enableARBAnnotations = params?.enableARBAnnotations || false;
+        this.enablePluralSuffixes = params?.enablePluralSuffixes || false;
+        this.emitComments = params?.emitComments || false;
+    }
+
+    async parseResource({ resource }) {
+        var segments = [];
+        var notes = {};
+        for (const [key, value] of Object.entries(flatten(resource))) {
+            if (key.indexOf("@") === -1) {
+                var seg = { sid: key, str: value };
+                notes[key] && (seg.notes = notes[key]);
```

Comment:
`??` is better here, in case you're so unlucky to have a falsy note.

Reply:
&& works here since I only do the assignment if notes is truthy